test/uk/gov/hmrc/payetaxcalculatorfrontend/model/TaxResultSpec.scala

```
@@ -0,0 +1,120 @@
+
+  "Extracting Tax Code from user response" should {
+
+    "return tax code that a user has provided" in {
+      extractTaxCode(QuickCalcAggregateInput(Some(UserTaxCode(true,Some("K452"))), None, None)) shouldBe "K452"
+    }
+
+    "return default tax code: 1150L if a user does not provide one" in {
+      extractTaxCode(QuickCalcAggregateInput(Some(UserTaxCode(false,None)), None, None)) shouldBe "1150L"
+    }
+  }
+
+  "Extracting Over65 answer from user response" should {
+
+    "return true if the response is user is over 65" in {
+      extractOver65(QuickCalcAggregateInput(None, Some(Over65(true)), None)) shouldBe "true"
+    }
+
+    "return false if the reponse is user is not over 65" in {
+      extractOver65(QuickCalcAggregateInput(None, Some(Over65(false)), None)) shouldBe "false"
+    }
+
+    //Todo unsure about this?
+    "return empty string if no response" in {
+      extractOver65(QuickCalcAggregateInput(None, None, None)) shouldBe ""
+    }
+  }
+
+  "Extracting Salary from user response" should {
+
+    "return if response provided is the Yearly Salary" in {
+      extractSalary(QuickCalcAggregateInput(None, None, Some(Yearly(20000)))) shouldBe 20000
+    }
+
+    "return if response provided is the Monthly Salary" in {
+      extractSalary(QuickCalcAggregateInput(None, None, Some(Monthly(2000)))) shouldBe 2000
+    }
+
+    "return if response provided is the Weekly Salary" in {
+      extractSalary(QuickCalcAggregateInput(None, None, Some(Weekly(200)))) shouldBe 200
+    }
+
+    "return if response provided is the Daily Salary" in {
+      extractSalary(QuickCalcAggregateInput(None, None, Some(Daily(20,0)))) shouldBe 20
+    }
+
+    "return if response provided is the Hourly Salary" in {
+      extractSalary(QuickCalcAggregateInput(None, None, Some(Hourly(2, 0)))) shouldBe 2
+    }
+
+    "return 0 if no response is provided" in {
+      extractSalary(QuickCalcAggregateInput(None, None, None)) shouldBe 0
```

We could throw an exception here instead as it should never be None at this point